### PR TITLE
Add list_fileset_files feature

### PR DESCRIPTION
**Pull request by AleksanderWWW**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/handler.py [hunk 1]
```diff
@@ -663,6 +664,14 @@ def fetch_files_list(self) -> List[ArtifactFileData]:
         """
         return self._pass_call_to_attr(function_name="fetch_files_list")
 
+    def fetch_fileset_files(self) -> List[FileEntry]:
```
Comment (Raalsky) [inline src/neptune/handler.py:667]: We already have `list_artifact_files`. Isin't the listing more proper naming here?

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 2]
```diff
@@ -850,6 +852,20 @@ def list_artifact_files(self, project_id: str, artifact_hash: str) -> List[Artif
         except HTTPNotFound:
             raise ArtifactNotFoundException(artifact_hash)
 
+    @with_api_exceptions_handler
+    def fetch_fileset_files(self, attribute, container_id) -> List[FileEntry]:
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_neptune_backend.py:856]: Typings for parameters

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 3]
```diff
@@ -850,6 +852,20 @@ def list_artifact_files(self, project_id: str, artifact_hash: str) -> List[Artif
...
+        try:
+            entries = (
+                self.leaderboard_client.api.lsFileSetAttribute(
+                    attribute=attribute, path=".", experimentId=container_id, holderType="run", **DEFAULT_REQUEST_KWARGS
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_neptune_backend.py:860]: Do we really need this `holderType`?

Comment (Raalsky) [inline src/neptune/internal/backends/hosted_neptune_backend.py:860]: Check `path_to_str`

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 4]
```diff
@@ -850,6 +852,20 @@ def list_artifact_files(self, project_id: str, artifact_hash: str) -> List[Artif
...
+                )
+                .response()
+                .result
+            )
+            return [FileEntry.from_dto(entry) for entry in entries]
+        except HTTPNotFound:
+            raise FileSetNotFound(f"No such attribute: '{attribute}'")
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_neptune_backend.py:867]: I think this should only pass an attribute path without any similar message.

File: src/neptune/internal/files/types.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+__all__ = ["FileEntry"]
```
Comment (Raalsky) [inline src/neptune/internal/files/types.py:1]: Let's place it in `neptune.api.dtos` or something like that

File: src/neptune/exceptions.py [hunk 6]
```diff
@@ -1176,3 +1178,28 @@ def __init__(self):
...
+        wait for all the requests sent by invoking:
+
+            {python}run.wait(){end}
+
+        Remember that each use of {python}wait{end} introduces a delay in code execution.
+
+        You may also want to check the following docs page:
+            - https://docs.neptune.ai/api/connection_modes
+
+        {correct}Need help?{end}-> https://docs.neptune.ai/getting_help.html
```
Comment (normandy7) [inline src/neptune/exceptions.py:1203]:
```suggestion
        {correct}Need help?{end}-> https://docs.neptune.ai/getting_help/
```

**Review by normandy7 — COMMENTED** [review]
I have a couple of questions:

1. What is the `sub_path` argument for?
2. If you track a directory, the size `None` is returned instead of the total size of all files (this info is available in the UI). Could it be available here as well?

I'll tweak the docstring a bit later.

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,7 @@
 ### Features
 - Users can pass neptune data directory path by env variable ([#1409](https://github.com/neptune-ai/neptune-client/pull/1409))
 - Filter S3 empty files and exclude metadata from computing of file's hash for new version of artifacts ([#1421](https://github.com/neptune-ai/neptune-client/pull/1421))
+- Added `list_fileset_files()` method to list files and directories contained in `FileSet` field ([#1412](https://github.com/neptune-ai/neptune-client/pull/1412))
```
Comment (Raalsky) [inline CHANGELOG.md:6]: Move to `1.6.0` (unreleased)

File: src/neptune/exceptions.py [hunk 8]
```diff
@@ -1176,3 +1178,28 @@ def __init__(self):
...
         super().__init__(message)
+
+
+class FileSetNotFound(NeptuneException):
+    def __init__(self, attribute: str, sub_path: str):
+        message = """
+        {h1}
+        ----MetadataInconsistency----------------------------------------------------------------------
+        {end}
+        Attribute {python}{path}{end} was not found.
```
Comment (Raalsky) [inline src/neptune/exceptions.py:1189]: `sub_path`? Or even better rename parameter to `path`?

**Review by Raalsky — DISMISSED** [review]

**Review by Raalsky — APPROVED** [review]

**Pull request merged**